### PR TITLE
Optionally provide notificationTopic instead of notificationEmail

### PR DESCRIPTION
**Pull request by emilymcafee**
Fixes https://github.com/mapbox/ecs-watchbot/issues/97

### CODE REVIEW

File: lib/template.js [hunk 1]
```diff
@@ -29,6 +29,8 @@ var table = require('@mapbox/watchbot-progress').table;
  * container that will be deployed. Defaults to the installed version
  * @param {string|ref} options.notificationEmail - an email address to receive
  * notifications when processing fails.
+ * @param {string|ref} options.notificationTopic - an ARN of the SNS topic to receive
```
Comment (rclark) [inline lib/template.js]: `[]` around the `options.notificationTopic` and `options.notificationEmail` to indicate that it they're optional

File: lib/template.js [hunk 2]
```diff
@@ -154,7 +156,10 @@ module.exports = function(options) {
   if (options.reduce) reduce(prefixed, resources, options, references);
   var mounts = mount(options.mounts);
 
-  resources[prefixed('NotificationTopic')] = {
+  if (options.notificationTopic && options.notificationEmail) throw new Error('Cannot provide both notificationTopic and notificationEmail.');
```
Comment (rclark) [inline lib/template.js]: I think you could allow both to be specified, and should assert that at least one is given. In the event that both exist, you'd just have 2 AlarmActions in the array

Comment (emilymcafee) [inline lib/template.js]: In the event that both exist, it's pretty straightforward to have 2 AlarmActions, but we'd need to do some re-writing in https://github.com/mapbox/ecs-watchbot/blob/master/lib/notifications.js to accept an array of topics, yeah?

Comment (rclark) [inline lib/template.js]: Ok, if you don't think that's worthwhile then 🚢

**Comment by emilymcafee** [comment]
I went down the road of implementing multiple notification topics, but we need this to be a string so that [it can be passed into the worker container's environment](https://github.com/mapbox/ecs-watchbot/blob/39e743f5742e0bd956e6bb1fbff229cf491aa02c/lib/template.js#L421). Punting and requiring one or the other in this iteration.

**Pull request merged**